.github/workflows/dev_ci_cd.yml

```diff
@@ -0,0 +1,37 @@
+name: Dev CIF API CI/CD
+
+on:
+  push:
+    branches: [ "bugfix/pytest" ]
+
+permissions:
+  contents: read
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: ["3.10"]
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install Linux dependencies
+      run: |
+        sudo apt update
+        sudo apt install -y gdal-bin libgdal-dev
+    - name: Install Packages
```

Comment:
Have you tried using this to set up the conda environment.yml file instead of having a different requirements.txt?